mmv1/third_party/terraform/tpgresource/utils.go

```diff
@@ -97,12 +99,49 @@ func GetProjectFromDiff(d *schema.ResourceDiff, config *transport_tpg.Config) (s
 	if ok {
 		return res.(string), nil
 	}
+	if d.GetRawConfig().GetAttr("project") == cty.UnknownVal(cty.String) {
```

Comment:
This handles the case where a project is interpolated based on a future resource, right?

Reply:
Yes, the rawconfig returns cty.unknownval for when a user entered a reference, and returns a cty.nullval for if it was actually unentered entirely.